backend/test/service_openai/test_ai_models.py

```diff
@@ -0,0 +1,176 @@
+        assert AIModel.get_token_param_name("o1") == "max_completion_tokens"
+        assert AIModel.get_token_param_name("o1-mini") == "max_completion_tokens"
+        assert AIModel.get_token_param_name("gpt-4o") == "max_completion_tokens"
+
+        # Test with enum values
+        assert AIModel.get_token_param_name(AIModel.O3_MINI) == "max_completion_tokens"
+        assert AIModel.get_token_param_name(AIModel.GPT_4O) == "max_completion_tokens"
+
+        # Test models that use max_tokens
+        assert AIModel.get_token_param_name("gpt-4") == "max_tokens"
+        assert AIModel.get_token_param_name("gpt-3.5-turbo") == "max_tokens"
+        assert AIModel.get_token_param_name("gpt-4-turbo") == "max_tokens"
+
+        # Test with enum values
+        assert AIModel.get_token_param_name(AIModel.GPT_4) == "max_tokens"
+        assert AIModel.get_token_param_name(AIModel.GPT_3_5_TURBO) == "max_tokens"
+
+        # Test models that aren't in our enum but should match patterns
+        assert AIModel.get_token_param_name("gpt-3.5-turbo-16k") == "max_tokens"
+        assert AIModel.get_token_param_name("o3") == "max_completion_tokens"
+        assert (
+            AIModel.get_token_param_name("gpt-4o-2024-05-13") == "max_completion_tokens"
+        )
+
+        # Test with unknown model - should default to max_tokens
+        assert AIModel.get_token_param_name("unknown-model") == "max_tokens"
+
+    def test_get_unsupported_parameters(self):
+        """
+        Test getting unsupported parameters for different models.
+
+        This test verifies that the correct list of unsupported
+        parameters is returned for different models.
+        """
+        # Test models with known unsupported parameters
+        assert "temperature" in AIModel.get_unsupported_parameters("o3-mini")
+        assert "top_p" in AIModel.get_unsupported_parameters("o1")
+        assert "parallel_tool_calls" in AIModel.get_unsupported_parameters("o1-mini")
+
+        # Test using enum values
+        assert "temperature" in AIModel.get_unsupported_parameters(AIModel.O3_MINI)
+        assert "top_p" in AIModel.get_unsupported_parameters(AIModel.O1)
+        assert "parallel_tool_calls" in AIModel.get_unsupported_parameters(
+            AIModel.O1_MINI
+        )
+
+        # Test models with pattern matching (o-series)
+        assert "temperature" in AIModel.get_unsupported_parameters("o3-2024-05-13")
+        assert "top_p" in AIModel.get_unsupported_parameters("o1-preview")
+
+        # Test models that should support all parameters
+        assert len(AIModel.get_unsupported_parameters("gpt-4")) == 0
+        assert len(AIModel.get_unsupported_parameters("gpt-3.5-turbo")) == 0
+        assert len(AIModel.get_unsupported_parameters(AIModel.GPT_4_TURBO)) == 0
+
+        # Test with unknown model - should default to empty list
+        assert len(AIModel.get_unsupported_parameters("unknown-model")) == 0
```

Comment:
Writing a test to check that an enum value is a supported value is not useful, delete.   I'll add some style guide stuff for what I want in unit tests.  They ought to test _functionality_, preferably for a single python function.